### PR TITLE
Inaugural cell users and roles. (SYN-3273)

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: synapse/lib/cell.py [hunk 1]
```diff
@@ -51,6 +51,67 @@
 Base classes for the synapse "cell" microservice architecture.
 '''
 
+_inagural_users_schema = {
```
Comment (vEpiphyte) [inline synapse/lib/cell.py:54]: TODO: Embed this directly into the cell config

File: synapse/lib/cell.py [hunk 2]
```diff
@@ -852,6 +918,12 @@ async def __anit__(self, dirn, conf=None, readonly=False, parent=None):
 
             self.inaugural = True
 
+            # Validate our _inaugural config PRIOR to any
+            # action which creates the cell.guild file.
+            iusers = self.conf.get('_inaugural')
```
Comment (vEpiphyte) [inline synapse/lib/cell.py:923]: This will be removed when the schema is embedded into the config definition.

File: synapse/lib/cell.py [hunk 3]
```diff
@@ -2063,6 +2139,40 @@ async def _initCellHiveAuth(self):
         self.onfini(auth.fini)
         return auth
 
+    async def _initInauguralCellAuth(self):
+        auth = self.auth  # type: s_hiveauth.Auth
+        if self.inaugural:
+            idata = self.conf.get('_inaugural')
+            if idata:
```
Comment (invisig0th) [inline synapse/lib/cell.py:2146]: is not None

File: synapse/tests/test_lib_cell.py [hunk 4]
```diff
@@ -1287,3 +1288,97 @@ async def _slowFakeBackup(self, name=None, wait=True):
...
                 nodes = await core.nodes('test:str=ssl')
                 self.len(1, nodes)
+
+    async def test_inaugural_users(self):
+
+        conf = {
+            '_inaugural': {
+                'users': [
+                    {
+                        'name': 'foo@bar.mynet.com',
```
Comment (invisig0th) [inline synapse/tests/test_lib_cell.py:1298]: would probably be good to test email plumbing too

**Review by invisig0th — CHANGES_REQUESTED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**